modRlp/test/org/aion/rlp/UtilsTest.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2017-2018 Aion foundation.
+ *
+ *     This file is part of the aion network project.
+ *
+ *     The aion network project is free software: you can redistribute it
+ *     and/or modify it under the terms of the GNU General Public License
+ *     as published by the Free Software Foundation, either version 3 of
+ *     the License, or any later version.
+ *
+ *     The aion network project is distributed in the hope that it will
+ *     be useful, but WITHOUT ANY WARRANTY; without even the implied
+ *     warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *     See the GNU General Public License for more details.
+ *
+ *     You should have received a copy of the GNU General Public License
+ *     along with the aion network project source files.
+ *     If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Contributors:
+ *     Aion foundation.
+ */
+package org.aion.rlp;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import java.util.Arrays;
+import org.aion.base.util.Hex;
+import org.junit.Test;
+
+public class UtilsTest {
+
+    @Test
+    public void testHexEncode_wNull() {
+        assertThat(Utils.hexEncode(null)).isNull();
+    }
+
+    @Test
+    public void testHexEncode_wSingleByte() {
+        for (byte b : Utils.encodingTable) {
+            byte[] input = new byte[] {b};
+            assertThat(Utils.hexEncode(input)).isEqualTo(Hex.encode(input));
+        }
+    }
+
+    @Test
+    public void testHexEncode_wString() {
+        String value = "1234567890abcdef";
```

Comment:
Not directly related to your code, but I was experimenting this with test and I added an 'a' to this string, so value = "1234567890abcdefa" and the Hex.decode(value) throws an exception.  Is that expected?

Reply:
`Hex.decode(value)` works only with even length strings because one byte is encoded to two characters.